[SPARK-29465][YARN][WEBUI] Adding Check to not to set UI port (spark.ui.port) property if mentioned explicitly

rajatahujaatinmobi opened the pull request:
## What changes were proposed in this pull request?
When a Spark Job launched in Cluster mode with Yarn, Application Master sets spark.ui.port port to 0 which means Driver's web UI gets any random port even if we want to explicitly set the Port range for Driver's Web UI

## Why are the changes needed?
We access Spark Web UI via Knox Proxy, and there are firewall restrictions due to which we can not access Spark Web UI since Web UI port range gets random port even if we set explicitly.

This Change will check if there is a specified port range explicitly mentioned so that it does not assign a random port.

## Does this PR introduce any user-facing change?
No

## How was this patch tested?
Local Tested.

Commit:
[SPARK-32039][YARN][WEBUI][Master] Adding Check to not to set UI port (spark.ui.port) property if mentioned explicitly

Commit:
[SPARK-32039][YARN][WEBUI][Master] Adding Check to not to set UI port (spark.ui.port) property if mentioned explicitly

AmplabJenkins commented:
Can one of the admins verify this patch?

AmplabJenkins commented:
Can one of the admins verify this patch?

dongjoon-hyun commented:
ok to test

dongjoon-hyun commented:
Thank you for moving to `master`, @rajatahujaatinmobi .

SparkQA commented:
**[Test build #124342 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/124342/testReport)** for PR 28880 at commit [`ce4b6ea`](https://github.com/apache/spark/commit/ce4b6ea7a4f43b776a57dec348f0265b07c6ca23).

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder-K8s/28961/
Test PASSed.

dongjoon-hyun commented on resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala:
```diff
@@ -211,9 +211,11 @@ private[spark] class ApplicationMaster(
+        // so we don't conflict with other spark processes running on the same box
+        if (System.getProperty(UI_PORT.key) != null) {
+          System.setProperty(UI_PORT.key, "0")
+        }
```
Hi, @rajatahujaatinmobi . While reviewing this PR, I noticed that SPARK-29465 is a previous one for this issue.

tgravescs replied:
yes, please see the discussions there, I think a range of ports would be much better. A single port set on a multi-tenant yarn cluster could easily end up with conflicts. What requirements do you have from Knox, a certain range or ports?
Normally on yarn I would expect you to use the resource manager web ui proxy on a secure cluster so you wouldn't be accessing the UI directly.

rajatahujaatinmobi replied:
Correct me if I am wrong

As per the discussions that UI port is always get assigned in the range based on spark.port.maxRetries. So we can always mention the starting port and afterward, it will try to start service based in between (starting port, starting port +spark.port.maxRetries). So we are including the range implicitly. 

Copying the comment
``other port types were spawned in the range (port_mentioned to port_mentioned+spark.port.maxRetries) except the UI port which started on random port.``

We have a Knox proxy that accesses UI via RM link. And Knox proxy has a certain range of ports that it can access on the cluster which is (18000-18159). Unless we start web UI on that range, we can not access the UI. and we can achieve above the range if we set the following properties
spark.ui.port=18000
spark.port.maxRetries=159

Hence setting up a spark.ui.port as starting port is enough instead of setting up a random port number in the range.

Ngone51 replied:
yea, I think yet Spark could utilize the combination of `spark.ui.port` and `spark.port.maxRetries` to perform "range port".

What else the  "range port" do you refer to? @tgravescs

tgravescs replied:
ah yes you are correct you can get range that way. the ports originally wouldn't do the retries or at least we didn't allow configuring maxRetries.

rajatahujaatinmobi replied:
I blocked a set of Port range from 18000 to 18050 and configure web or to start at 18008 but it eventually Started at 18051 

**Properties Set: spark.ui.port=18018  spark.port.maxRetries=60**
Successful Case:
20/06/23 21:55:30 WARN util.Utils: Service 'SparkUI' could not bind on port 18035. Attempting port 18036.
20/06/23 21:55:30 WARN util.Utils: Service 'SparkUI' could not bind on port 18036. Attempting port 18037.
20/06/23 21:55:30 WARN util.Utils: Service 'SparkUI' could not bind on port 18037. Attempting port 18038.
20/06/23 21:55:30 WARN util.Utils: Service 'SparkUI' could not bind on port 18038. Attempting port 18039.
20/06/23 21:55:30 WARN util.Utils: Service 'SparkUI' could not bind on port 18039. Attempting port 18040.
20/06/23 21:55:30 WARN util.Utils: Service 'SparkUI' could not bind on port 18040. Attempting port 18041.
20/06/23 21:55:30 WARN util.Utils: Service 'SparkUI' could not bind on port 18041. Attempting port 18042.
20/06/23 21:55:30 WARN util.Utils: Service 'SparkUI' could not bind on port 18042. Attempting port 18043.
20/06/23 21:55:30 WARN util.Utils: Service 'SparkUI' could not bind on port 18043. Attempting port 18044.
20/06/23 21:55:30 WARN util.Utils: Service 'SparkUI' could not bind on port 18044. Attempting port 18045.
20/06/23 21:55:30 WARN util.Utils: Service 'SparkUI' could not bind on port 18045. Attempting port 18046.
20/06/23 21:55:30 WARN util.Utils: Service 'SparkUI' could not bind on port 18046. Attempting port 18047.
20/06/23 21:55:30 WARN util.Utils: Service 'SparkUI' could not bind on port 18047. Attempting port 18048.
20/06/23 21:55:30 WARN util.Utils: Service 'SparkUI' could not bind on port 18048. Attempting port 18049.
20/06/23 21:55:30 WARN util.Utils: Service 'SparkUI' could not bind on port 18049. Attempting port 18050.
20/06/23 21:55:30 WARN util.Utils: Service 'SparkUI' could not bind on port 18050. Attempting port 18051.
20/06/23 21:55:30 INFO server.AbstractConnector: Started ServerConnector@7ba63fe5{HTTP/1.1,[http/1.1]}{0.0.0.0:18051}
**20/06/23 21:55:30 INFO util.Utils: Successfully started service 'SparkUI' on port 18051.
And  Job succeeded** 

Failure Case 
**Properties Set: spark.ui.port=18018  spark.port.maxRetries=6**

20/06/23 22:16:15 WARN util.Utils: Service 'SparkUI' could not bind on port 18018. Attempting port 18019.
20/06/23 22:16:15 WARN util.Utils: Service 'SparkUI' could not bind on port 18019. Attempting port 18020.
20/06/23 22:16:15 WARN util.Utils: Service 'SparkUI' could not bind on port 18020. Attempting port 18021.
20/06/23 22:16:15 WARN util.Utils: Service 'SparkUI' could not bind on port 18021. Attempting port 18022.
20/06/23 22:16:15 WARN util.Utils: Service 'SparkUI' could not bind on port 18022. Attempting port 18023.
20/06/23 22:16:15 WARN util.Utils: Service 'SparkUI' could not bind on port 18023. Attempting port 18024.
**20/06/23 22:16:15 ERROR ui.SparkUI: Failed to bind SparkUI
java.net.BindException: Address already in use: Service 'SparkUI' failed after 6 retries (starting from 18018)! Consider explicitly setting the appropriate port for the service 'SparkUI' (for example spark.ui.port for SparkUI) to an available port or increasing spark.port.maxRetries.
	at sun.nio.ch.Net.bind0(Native Method)
Job Failed ...**

@tgravescs @Ngone51

Ngone51 replied:
I don't get your point here. For the successful case, I only see the port range is 18018 ~ 18078 and the port 18051 is valid. Where are those ports 18000, 18050, 18008?

rajatahujaatinmobi replied:
I blocked ports from **18000 to 18050 s**o now spark.ui.port can not get port b/w that range. But in Spark properties, I mentioned  **spark.ui.port as 18018 and spark.port.maxRetries=60** so now spark service will start trying from 18018 port until **18018 + 60** in an increment way but ports are blocked until 18050 so spark.ui.port will pick up **18051** port.

Ngone51 replied:
I see. This is what we expected, right?

rajatahujaatinmobi replied:
Yes, that's expected. So I think we are good to go unless someone else has any other doubts.
@Ngone51 @tgravescs

tgravescs replied:
thanks for testing that, please update the comment like my below comment requests.

dongjoon-hyun commented:
@rajatahujaatinmobi . This is a partial revert of SPARK-3627 and has been discussed before. Please see SPARK-29465.
The last advice was `port range` might be acceptable solution.

SparkQA commented:
**[Test build #124342 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/124342/testReport)** for PR 28880 at commit [`ce4b6ea`](https://github.com/apache/spark/commit/ce4b6ea7a4f43b776a57dec348f0265b07c6ca23).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/124342/
Test PASSed.

rajatahujaatinmobi commented:
> @rajatahujaatinmobi . This is a partial revert of [SPARK-3627](https://issues.apache.org/jira/browse/SPARK-3627) and has been discussed before. Please see [SPARK-29465](https://issues.apache.org/jira/browse/SPARK-29465).
> The last advice was `port range` might be acceptable solution.


If spark.ui.port is Set then it will start from that port and will try web ui until spark.port.maxRetries and its default value is 16. So it takes care of range. 
```

spark.port.maxRetries | 16 | Maximum number of retries when binding to a port before giving up.     When a port is given a specific value (non 0), each subsequent retry will     increment the port used in the previous attempt by 1 before retrying. This     essentially allows it to try a range of ports from the start port specified     to port + maxRetries.
-- | -- | --

dongjoon-hyun commented:
Is that the same comment with https://issues.apache.org/jira/browse/SPARK-29465?focusedCommentId=16952608&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-16952608 ?

dongjoon-hyun commented:
cc @tgravescs

Ngone51 commented on resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala:
```diff
@@ -211,9 +211,11 @@ private[spark] class ApplicationMaster(
-        System.setProperty(UI_PORT.key, "0")
+        // Set the web ui port to be ephemeral for yarn if not set explicitly
+        // so we don't conflict with other spark processes running on the same box
+        if (System.getProperty(UI_PORT.key) != null) {
```
Shouldn't it be `==`?

rajatahujaatinmobi replied:
Yes, changing the If condition.

Ngone51 commented:
Seems like the change is firstly introduced in https://github.com/apache/spark/commit/c6de982be69cd50e66375cfea3d6c3267a01583b rather than SPARK-3627.

rajatahujaatinmobi commented:
> Is that the same comment with https://issues.apache.org/jira/browse/SPARK-29465?focusedCommentId=16952608&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-16952608 ?

Yes.

HyukjinKwon commented:
cc @tgravescs FYI

Commit:
[SPARK-32039][YARN][WEBUI][Master] Adding Check to not to set UI port (spark.ui.port) property if mentioned explicitly

Commit:
[SPARK-32021][SQL] Increase precision of seconds and fractions of `make_interval`

Change precision of seconds and its fraction from 8 to 18 to be able to construct intervals of max allowed microseconds value (long).

To improve UX of Spark SQL.

Yes

- Add tests to IntervalExpressionsSuite
- Add an example to the `MakeInterval` expression
- Add tests to `interval.sql`

Closes #28873 from MaxGekk/make_interval-sec-precision.

Authored-by: Max Gekk <max.gekk@gmail.com>
Signed-off-by: Dongjoon Hyun <dongjoon@apache.org>

[SPARK-31957][SQL] Cleanup hive scratch dir for the developer api startWithContext

Comparing to the long-running ThriftServer via start-script, we are more likely to hit the issue https://issues.apache.org/jira/browse/HIVE-10415 / https://issues.apache.org/jira/browse/SPARK-31626 in the developer API `startWithContext`

This PR apply SPARK-31626 to the developer API `startWithContext`

Fix the issue described in  SPARK-31626

Yes, the hive scratch dir will be deleted if cleanup is enabled for calling `startWithContext`

new test

Closes #28784 from yaooqinn/SPARK-31957.

Authored-by: Kent Yao <yaooqinn@hotmail.com>
Signed-off-by: Dongjoon Hyun <dongjoon@apache.org>

[SPARK-32039][YARN][WEBUI][Master] Adding Check to not to set UI port (spark.ui.port) property if mentioned explicitly

[SPARK-32039][YARN][WEBUI][Master] Adding Check to not to set UI port (spark.ui.port) property if mentioned explicitly

Merge branch 'ahujarajat261/SPARK-32039-change-yarn-webui-port-range-with-property-latest-spark' of https://github.com/rajatahujaatinmobi/spark-1 into ahujarajat261/SPARK-32039-change-yarn-webui-port-range-with-property-latest-spark

SparkQA commented:
**[Test build #124349 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/124349/testReport)** for PR 28880 at commit [`10b9f7e`](https://github.com/apache/spark/commit/10b9f7e3bbe3685ccd27e2d71ac50e40b17588f9).

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder-K8s/28968/
Test PASSed.

SparkQA commented:
**[Test build #124349 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/124349/testReport)** for PR 28880 at commit [`10b9f7e`](https://github.com/apache/spark/commit/10b9f7e3bbe3685ccd27e2d71ac50e40b17588f9).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/124349/
Test PASSed.

tgravescs commented on resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala:
```diff
@@ -211,9 +211,11 @@ private[spark] class ApplicationMaster(
-        // other spark processes running on the same box
-        System.setProperty(UI_PORT.key, "0")
+        // Set the web ui port to be ephemeral for yarn if not set explicitly
+        // so we don't conflict with other spark processes running on the same box
```
lets update the comment to say if explicitly set, it acts as a port range based on spark.port.maxRetries so make sure that is sufficiently large for your setup to avoid port conflicts.

tgravescs reviewed (commented):
I believe this will be fine as I don't think we set the default values as properties unless its explicitly set, but please test it. And test both cases where it isn't set it should go to 0 and when it is set it should iterate through the range with retries.  From the bug of != vs == I'm assuming this code wasn't run on an actual yarn cluster, so please do that.

SparkQA commented:
**[Test build #124559 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/124559/testReport)** for PR 28880 at commit [`b42c3ed`](https://github.com/apache/spark/commit/b42c3ed7498bad6d3607978c0c19979828da9ef0).

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder-K8s/29180/
Test PASSed.

SparkQA commented:
**[Test build #124559 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/124559/testReport)** for PR 28880 at commit [`b42c3ed`](https://github.com/apache/spark/commit/b42c3ed7498bad6d3607978c0c19979828da9ef0).
 * This patch **fails Scala style tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/124559/
Test FAILed.

Commit:
[SPARK-29465][YARN][WEBUI] Adding Check to not to set UI port (spark.ui.port) property if mentioned explicitly

Commit:
[SPARK-29465][YARN][WEBUI] Adding Check to not to set UI port (spark.ui.port) property if mentioned explicitly

SparkQA commented:
**[Test build #124562 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/124562/testReport)** for PR 28880 at commit [`8fe0aa9`](https://github.com/apache/spark/commit/8fe0aa98a18edb579b7ee214b6c6dd9ba69b395b).

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder-K8s/29183/
Test PASSed.

SparkQA commented:
**[Test build #124562 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/124562/testReport)** for PR 28880 at commit [`8fe0aa9`](https://github.com/apache/spark/commit/8fe0aa98a18edb579b7ee214b6c6dd9ba69b395b).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/124562/
Test PASSed.

Ngone51 commented:
LGTM

rajatahujaatinmobi reviewed (commented):
> LGTM

How can we go ahead and merge it with master ?

tgravescs commented:
rekicked the checks, once they pass I can commit

rajatahujaatinmobi reviewed (commented):
> rekicked the checks, once they pass I can commit

continuous-integration/appveyor/pr  is failing. Not sure why is that so?

rajatahujaatinmobi commented:
> > rekicked the checks, once they pass I can commit
> 
> continuous-integration/appveyor/pr is failing. Not sure why is that so?



> rekicked the checks, once they pass I can commit
@tgravescs  can you rerun the tests and commit/merge with master if it seems okay? I can not rerun the tests

tgravescs commented:
test this please

SparkQA commented:
**[Test build #124664 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/124664/testReport)** for PR 28880 at commit [`8fe0aa9`](https://github.com/apache/spark/commit/8fe0aa98a18edb579b7ee214b6c6dd9ba69b395b).

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder-K8s/29280/
Test PASSed.

SparkQA commented:
**[Test build #124664 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/124664/testReport)** for PR 28880 at commit [`8fe0aa9`](https://github.com/apache/spark/commit/8fe0aa98a18edb579b7ee214b6c6dd9ba69b395b).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/124664/
Test PASSed.

Commit:
Rebasing with Upstream

Commit:
[SPARK-29465][YARN][WEBUI] Adding Check to not to set UI port (spark.ui.port) property if mentioned explicitly

Commit:
[SPARK-29465][YARN][WEBUI] Adding Check to not to set UI port (spark.ui.port) property if mentioned explicitly

SparkQA commented:
**[Test build #124685 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/124685/testReport)** for PR 28880 at commit [`24ecfcf`](https://github.com/apache/spark/commit/24ecfcf1c06a4af8f15ce28e6cd9b559bba7643b).

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder-K8s/29298/
Test PASSed.

SparkQA commented:
**[Test build #124687 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/124687/testReport)** for PR 28880 at commit [`1896905`](https://github.com/apache/spark/commit/1896905d42ee71709fa0da0a80d84fe3eb026c6e).

SparkQA commented:
**[Test build #124687 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/124687/testReport)** for PR 28880 at commit [`1896905`](https://github.com/apache/spark/commit/1896905d42ee71709fa0da0a80d84fe3eb026c6e).
 * This patch **fails to generate documentation**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/124687/
Test FAILed.

SparkQA commented:
**[Test build #124685 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/124685/testReport)** for PR 28880 at commit [`24ecfcf`](https://github.com/apache/spark/commit/24ecfcf1c06a4af8f15ce28e6cd9b559bba7643b).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/124685/
Test PASSed.

rajatahujaatinmobi commented:
> test this please
Jenkins jobs seems to be doing fine
https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/124664/

> **[Test build #124685 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/124685/testReport)** for PR 28880 at commit [`24ecfcf`](https://github.com/apache/spark/commit/24ecfcf1c06a4af8f15ce28e6cd9b559bba7643b).
> 
>     * This patch passes all tests.
> 
>     * This patch merges cleanly.
> 
>     * This patch adds no public classes.

@tgravescs  Looks like all the test cases pass in the Jenkins Build. Shall we  go ahead and merge it ?

tgravescs commented:
this is odd, looks like those are all R tests for some reason failing. @dongjoon-hyun  have you seen that before?

rajatahujaatinmobi commented:
> this is odd, looks like those are all R tests for some reason failing. @dongjoon-hyun have you seen that before?

@dongjoon-hyun  Any updates as to how can we proceed further?

dongjoon-hyun commented:
For me, it looks passed with the following options, @tgravescs and @rajatahujaatinmobi .
```
-Phadoop-3.2 -Phive-2.3 -Pyarn -Dtest.exclude.tags=org.apache.spark.tags.ExtendedHiveTest yarn/test network-yarn/test
```

dongjoon-hyun reviewed (approved):
+1, LGTM. Thank you, @rajatahujaatinmobi , @tgravescs , @Ngone51 , @HyukjinKwon .
Merged to master for Apache Spark 3.1.

dongjoon-hyun commented:
Hi, @rajatahujaatinmobi . What is your Apache JIRA id? I need to assign SPARK-29465 to you.
